csrc/scheduler/no_op.cpp

```diff
@@ -39,9 +39,18 @@ bool allOutputsArePointerArithmetics(Fusion* fusion) {
   auto out_tvs = ir_utils::filterByType<TensorView>(fusion->outputs());
   return std::all_of(
       out_tvs.begin(), out_tvs.end(), [&analysis, fusion](TensorView* out) {
-        // Check out has an alias and out is not an inplace update target.
-        return analysis.getNearestAliasedIo(out) != nullptr &&
-            fusion->getOutputAlias(out).type != AllocationType::ReuseBuffer;
+        // Check out is not an inplace update target.
+        if (fusion->getOutputAlias(out).type == AllocationType::ReuseBuffer) {
+          return false;
+        }
+        auto val = analysis.getNearestAliasedIo(out);
```

Comment:
I think this part should be refactored later per our offline conversation regarding how to detect the source and replay meta operations.